cmd/eval/main.go

```diff
@@ -0,0 +1,239 @@
+		playerTwoId, err := strconv.Atoi(data[2])
+		if err != nil {
+			panic(err)
+		}
+		tourneyId, err := strconv.Atoi(data[5])
+		if err != nil {
+			panic(err)
+		}
+		lexicon := data[7]
+		if (playerOneId == *playerIdPtr || playerTwoId == *playerIdPtr) &&
+			(*tourneyIdPtr == -1 || tourneyId == *tourneyIdPtr) &&
+			gameid >= *minGameIdPtr && gameid <= *maxGameIdPtr {
+			gameids = append(gameids, gameid)
+			lexicons = append(lexicons, lexicon)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		panic(err)
+	}
+
+	eqlossData := &EqlossData{Items: []*EqlossItem{}}
+	totalEqloss := 0.0
+	for idx, gameid := range gameids {
+		lexicon := lexicons[idx]
+		gcgFilepath := "gcg"
+		downloadFile(gcgFilepath, fmt.Sprintf("https://www.cross-tables.com/annotated/selfgcg/%d/anno%d.gcg", gameid/100, gameid))
+		eqloss, err := getEquityLoss(gcgFilepath, lexicon, *playerNamePtr, gameid, eqlossData)
+		if err != nil {
+			panic(err)
+		}
+		totalEqloss += eqloss
+	}
+	fmt.Printf("Average Equity Loss: %.2f\nTotal Games: %d\n", totalEqloss/float64(len(gameids)), len(gameids))
+	fmt.Printf("Total Plays: %d\n", len(eqlossData.Items))
+
+	sort.Slice(eqlossData.Items, func(i, j int) bool {
+		return eqlossData.Items[i].Eqloss > eqlossData.Items[j].Eqloss
+	})
+
+	maxPlayLen := -1
+	for i := 0; i < 10 && i < len(eqlossData.Items); i++ {
+		if len(eqlossData.Items[i].YourPlay) > maxPlayLen {
+			maxPlayLen = len(eqlossData.Items[i].YourPlay)
+		}
+	}
+
+	fmt.Printf("\n\nTop %d Saddest Anime Moments:\n\n", *playLimitPtr)
+	fmt.Printf("%-"+strconv.Itoa(maxPlayLen)+"s | %s  | %s | %s\n", "Your Play", "Loss", "Walid?", "Best Play")
+	for i := 0; i < *playLimitPtr && i < len(eqlossData.Items); i++ {
+		wetoed := "Strong"
+		if eqlossData.Items[i].PhonyTilesReturned {
+			wetoed = "NO    "
+		}
+		fmt.Printf("%-"+strconv.Itoa(maxPlayLen)+"s | %.2f | %s | %s\n", eqlossData.Items[i].YourPlay, eqlossData.Items[i].Eqloss, wetoed, eqlossData.Items[i].BestPlay)
+	}
+}
+
+func getEquityLoss(filepath string, lexicon string, playerName string, gameid int, eqlossData *EqlossData) (float64, error) {
+	rules, err := game.NewBasicGameRules(&DefaultConfig, lexicon, board.CrosswordGameLayout, "english", game.CrossScoreAndSet, game.VarClassic)
+	if err != nil {
+		panic(err)
+	}
+
+	gameHistory, err := gcgio.ParseGCG(&DefaultConfig, filepath)
+	if err != nil {
+		panic(err)
+	}
+	p1Nickname := gameHistory.Players[0].Nickname
+	p2Nickname := gameHistory.Players[1].Nickname
+	if playerName != p1Nickname && playerName != p2Nickname {
+		panic(fmt.Sprintf("player %s not found in (%s, %s) for game %d", playerName, p1Nickname, p2Nickname, gameid))
+	}
+
+	gameHistory.ChallengeRule = pb.ChallengeRule_FIVE_POINT
+
+	g, err := game.NewFromHistory(gameHistory, rules, 0)
+	if err != nil {
+		panic(err)
+	}
+
+	totalEqloss := 0.0
+	numMoves := 0
+	history := g.History()
+	players := history.Players
+	for evtIdx, evt := range history.Events {
+		if players[evt.PlayerIndex].Nickname == playerName &&
```

Comment:
this will need to get rewritten to get the PlayerIndex but it's ok for now. We really need to do that migration on prod. Let's brainstorm about how to complete it. It's tough because it requires a big database migration and I don't want to do it multiple times.